### PR TITLE
fix: [CDS-76291]: add banner to prevent user to aid in setting up the barrier

### PR DESCRIPTION
**Pull request by Eric761**
### Summary

Ticket - https://harness.atlassian.net/browse/CDS-76291
- Added banner to prevent user to aid in setting up the barrier

#### Screenshots

Screenshot 2023-08-11 at 3 39 27 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 1]
```diff
@@ -92,6 +92,8 @@ barrierStep:
   barrierReference: Barrier Reference
   barrierReferenceRequired: Barrier Reference is required
   barrierReferencePlaceholder: Select or Enter Barrier Reference
+  helpText: When deploying interdependent services, coordinating the timing of component deployments is crucial. The Harness Barrier’s Step can be used to synchronize different stages and step groups in a Pipeline. These barriers work effectively when two or more parallel stages/step groups share the same barrier name, ensuring synchronized execution.
+  infoLink: https://developer.harness.io/docs/continuous-delivery/x-platform-cd-features/cd-steps/flow-control/synchronize-deployments-using-barriers/
```
Comment (rameshx) [inline src/modules/70-pipeline/strings/strings.en.yaml:96]: This should not be in strings. Define a constant in any util or constants file

File: src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.module.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+.infoContainer {
+  a {
+    color: var(--primary-7) !important;
+  }
+
+  span[data-icon='info'] {
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.module.scss:13]: Let's pass `className` to `<Icon />`

**Review by rameshx — CHANGES_REQUESTED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.module.scss (hunk 2 above)
Comment (Eric761) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.module.scss]: done

File: src/modules/70-pipeline/strings/strings.en.yaml (hunk 1 above)
Comment (Eric761) [reply src/modules/70-pipeline/strings/strings.en.yaml]: done @rameshx

**Review by rameshx — DISMISSED** [review]

**Review by pankajsingh-harness — DISMISSED** [review]

**Review by rameshx — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Pull request merged**